modules/loader-utils/src/lib/worker-utils/worker-pool.js

```diff
@@ -80,7 +88,7 @@ export default class WorkerPool {
     }
 
     // Create fresh worker if we haven't yet created the max amount of worker threads for this worker source
-    if (this.count < this.maxConcurrency) {
+    if (this.count < this.maxConcurrency || this.notCashWorkers) {
```

Comment:
We can remove this check but if `this.count >= this.maxConcurrency` we will get stuck just because we will wait for free worker to proceed. In this case I think we should decrease` this.count `every time when we remove worker after his job is done.

Reply:
Yes good point. Maybe the key is to still call ` this._startQueuedJob();` but not push back the worker on the queue.

Reply:
I see your point but in this case we will get stuck again because we will never destroy existing workers and `this.count` will not be decreased. Therefore I need somewhere decrease `this.count` and it can confuse. My suggestion is to save current behavior here.